server/src/main/java/org/elasticsearch/index/get/ShardGetService.java

```diff
@@ -267,7 +267,11 @@ private GetResult innerGetFetch(
                 if (false == needed.contains(entry.getKey())) {
                     continue;
                 }
-                List<Object> values = FetchPhase.processStoredField(mapperService::fieldType, entry.getKey(), entry.getValue());
+                MappedFieldType ft = mapperService.fieldType(entry.getKey());
+                if (ft == null) {
+                    continue;
```

Comment:
I'm not sure what you mean - the point here is that asking for a non-existing field won't throw an error, it will just be ignored.  We could re-work things to go via `mapperService.getMatchingFieldNames()` which would filter out non-existent fields for us and would have the extra advantage of resolving aliases, but perhaps that should be in a follow up, or as a larger move to support `fields` in `get`.